crates/core-new/src/processor.rs

```diff
@@ -0,0 +1,101 @@
+//! Processor is reponsible for running the module.
+//! This run is sound execution of the module.
+//! This will be most abstract layer of the processor.
+
+use std::collections::HashSet;
+
+use anyhow::Result;
+
+use hdp_primitives::task::ExtendedTask;
+use hdp_provider::{
+    evm::{AbstractProvider, AbstractProviderConfig, AbstractProviderResult},
+    key::FetchKeyEnvelope,
+};
+
+use crate::cairo_runner::{
+    input::run::RunnerInput,
+    run::{RunResult, Runner},
+};
+
+pub struct Processor {
+    runner: Runner,
+}
+
+impl Processor {
+    pub fn new() -> Self {
+        let runner = Runner::new();
+        Self { runner }
+    }
+
+    pub async fn process(
+        &self,
+        tasks: Vec<ExtendedTask>,
```

Comment:
this could be preprocess result object as an argument to process function

Reply:
yes indeed resolved